### PR TITLE
Ensuring we pick the largest matching key from nameSizeMapping

### PR DESCRIPTION
**Pull request by ramitha**
- This was causing errors when we had names like
    "nameSizeMapping": {
      "xxsmall": {
        "width": 56,
        "height": 14
      },
      "xsmall": {
        "width": 84,
        "height": 21
      },
      "small":{
        "width": 110,
        "height": 28
      }
   }

### CODE REVIEW

File: packages/@icon-magic/generate/src/plugins/svg-generate.ts [hunk 1]
```diff
@@ -79,8 +79,13 @@ export const svgGenerate: GeneratePlugin = {
         // get the size from the mapping that is passed in. This is a pattern
         // matching of the key and not necessarily the key itself
         let flavorSize;
+        let matchedLength = 0;
         for (const key in nameSizeMapping) {
-          if (flavorName.match(key)) flavorSize = nameSizeMapping[key];
+          const regExMatch = flavorName.match(key);
+          if (regExMatch && matchedLength < regExMatch.length) {
```
Comment (oa495) [inline packages/@icon-magic/generate/src/plugins/svg-generate.ts]: can you add a comment giving more context to what's happening here?

File: packages/@icon-magic/generate/src/plugins/svg-to-raster.ts [hunk 2]
```diff
@@ -64,8 +64,13 @@ export const svgToRaster: GeneratePlugin = {
 
         // get the size from the mapping that is passed in. This is a pattern
         // matching of the key and not necessarily the key itself
+        let matchedLength = 0;
         for (const key in nameSizeMapping) {
-          if (flavorName.match(key)) sizeFromMapping = nameSizeMapping[key];
+          const regExMatch = flavorName.match(key);
+          if (regExMatch && matchedLength < regExMatch.length) {
```
Comment (oa495) [inline packages/@icon-magic/generate/src/plugins/svg-to-raster.ts]: same here

File: packages/@icon-magic/generate/src/plugins/svg-generate.ts (hunk 1 above)
Comment (ramitha) [reply packages/@icon-magic/generate/src/plugins/svg-generate.ts]: Done

File: packages/@icon-magic/generate/src/plugins/svg-to-raster.ts (hunk 2 above)
Comment (ramitha) [reply packages/@icon-magic/generate/src/plugins/svg-to-raster.ts]: Done

**Pull request merged**